sbot/motor_board.py

```diff
@@ -0,0 +1,267 @@
+"""The motor board module provides an interface to the motor board firmware over serial."""
+from __future__ import annotations
+
+import atexit
+import logging
+from enum import IntEnum
+from types import MappingProxyType
+
+from serial.tools.list_ports import comports
+
+from .exceptions import BoardDisconnectionError, IncorrectBoardError
+from .logging import log_to_debug
+from .serial_wrapper import SerialWrapper
+from .utils import (
+    Board, BoardIdentity, float_bounds_check,
+    get_USB_identity, map_to_float, map_to_int,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class MotorPower(IntEnum):
+    """Special values for motor power."""
+    BRAKE = 0
+    COAST = -1024  # A value outside the allowable range
+
+
+class MotorBoard(Board):
+    """
+    A class representing the motor board interface.
+
+    This class is intended to be used to communicate with the motor board over serial
+    using the text-based protocol added in version 4.4 of the motor board firmware.
+
+    :param serial_port: The serial port to connect to.
+    :param initial_identity: The identity of the board, as reported by the USB descriptor.
+    """
+    __slots__ = ('_serial', '_identity', '_motors')
+
+    @staticmethod
+    def get_board_type() -> str:
+        """
+        Return the type of the board.
+
+        :return: The literal string 'MCv4B'.
+        """
+        return 'MCv4B'
+
+    def __init__(
+        self,
+        serial_port: str,
+        initial_identity: BoardIdentity | None = None,
+    ) -> None:
+        if initial_identity is None:
+            initial_identity = BoardIdentity()
+        self._serial = SerialWrapper(serial_port, 115200, identity=initial_identity)
```

Comment:
nit: Not magic number?

Reply:
05a96aae1e73a3fe1a7ff92f3360334b24f3095e